File: source/253.swift
import UIKit

 public class Interval {
   public var start: Int
   public var end: Int
   public init(_ start: Int, _ end: Int) {
     self.start = start
     self.end = end
   }
 }

class Solution {
    func minMeetingRooms(_ intervals: [Interval]) -> Int {
        
        let starts = intervals.map { $0.start }.sorted()
        let ends   = intervals.map { $0.end }.sorted()
        
        var count = 0, i = 0, j = 0, res = 0
        
        while i < starts.count && j < ends.count {
            
            if starts[i] < ends[j] {
                i += 1
                count += 1
            } else {
                j += 1
                count -= 1
            }
            
            res = max(res, count)
        }
        
        return res
    }
}

File: source/253.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
#include <deque>
#include <queue>
#include <set>
#include <map>
#include <stack>
#include <cmath>
#include <numeric>
#include <unordered_map>
#include <sstream>
#include <unordered_set>

using namespace std;

struct Interval {
    int start;
    int end;
    Interval() : start(0), end(0) {}
    Interval(int s, int e) : start(s), end(e) {}
};

bool comp(Interval a, Interval b) {
    return a.start < b.start || (a.start == b.start && a.end < b.end);
}

class Solution {
public:
    int minMeetingRooms(vector<Interval>& intervals) {
        
        if (intervals.size() == 0) return 0;
        
        priority_queue<int, vector<int>, greater<int>> q;

        sort(intervals.begin(), intervals.end(), comp);
        q.push(intervals[0].end);
        
        for (int i = 1; i < intervals.size(); i++) {
            if (q.top() < intervals[i].start) {
                q.pop();
            }
            
            q.push(intervals[i].end);
        }
        
        return q.size();
    }
};